### PR TITLE
[speechx]add reset of decoder&frontend

### PR DESCRIPTION
**Pull request by SmileGoat**
### PR types

[New features]
### Describe

add reset of decoder & frontend

### CODE REVIEW

**Comment by SmileGoat** [comment]
vargrind & run.sh 稍后加，感觉代码还要再测试一下。

File: speechx/speechx/decoder/ctc_beam_search_decoder.cc [hunk 1]
```diff
@@ -202,15 +206,15 @@ int32 CTCBeamSearch::SearchOneChar(
     size_t beam_size = opts_.beam_size;
     const auto& c = log_prob_idx.first;
     const auto& log_prob_c = log_prob_idx.second;
-    size_t prefixes_len = std::min(prefixes.size(), beam_size);
+    size_t prefixes__len = std::min(prefixes_.size(), beam_size);
```
Comment (zh794390558) [inline speechx/speechx/decoder/ctc_beam_search_decoder.cc]: prefixes__len

File: speechx/examples/feat/run.sh [hunk 2]
```diff
@@ -18,10 +20,9 @@ if [ ! -d ../paddle_asr_model ]; then
...
 fi
 
-    echo "utt1 " $PWD/../paddle_asr_model/BAC009S0764W0290.wav > ../paddle_asr_model/wav.scp
 model_dir=../paddle_asr_model
 feat_wspecifier=./feats.ark
 cmvn=./cmvn.ark
 
 # 3. run feat
-../../build/examples/feat/linear_spectrogram_main --wav_rspecifier=scp:$model_dir/wav.scp --feature_wspecifier=ark:$feat_wspecifier --cmvn_write_path=$cmvn
+../../fc_patch/valgrind-build/bin/valgrind --tool=memcheck --track-origins=yes --leak-check=full --show-leak-kinds=all linear_spectrogram_main --wav_rspecifier=scp:$model_dir/wav.scp --feature_wspecifier=ark,t:$feat_wspecifier --cmvn_write_path=$cmvn
```
Comment (SmileGoat) [inline speechx/examples/feat/run.sh]: fixing

**Review by zh794390558 — APPROVED** [review]
LGTM

**Pull request merged**